### ISSUE TITLE
On Tester view, C++, when double clicking a test it navigates to a CMake file

### ISSUE THREAD

**Issue by yoavmil** [microsoft/vscode-cmake-tools#3512]
Type: <b>Bug</b>

On Tester view, using GTest C++, when double clicking a test it navigates to a CMake file "C:\Program Files\CMake\share\cmake-3.25\Modules\GoogleTest.cmake". Instead, it shoud navigate to the code of the test. This also happens when clicking the "Go-To-Test" button. This forces me to navigate to the test manually which could take time.

Maybe it's relevant, to build the tests I use CMake gtest_add_tests(TARGET target_name).

(Related, when clicking the test Debug button doesn't actually run this specific test in debug mode)

Extension version: 1.17.7
VS Code version: Code 1.85.1 (0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2, 2023-12-13T09:49:37.021Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz (16 x 3600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.92GB (5.13GB free)|
|Process Argv|--crash-reporter-id 9d19a473-c68b-42a9-b9c5-cf4377236ced|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263:30899288
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
282f8724:30602487
f6dab269:30613381
a9j8j154:30646983
showlangstatbar:30737416
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixt:30902429
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
aa_t_chat:30882232
cp7184c:30925681

```

</details>

**Comment by gcampbell-msft** [comment]
@yoavmil To help us investigate this, could you provide a small repro project so we could reproduce and investigate this issue? 

Thanks!

**Comment by yoavmil** [comment]
https://github.com/yoavmil/cmake_tests_example
There you go.
Thanks.

**Comment by bertbuchholz** [comment]
> On Tester view, using GTest C++, when double clicking a test it navigates to a CMake file "C:\Program Files\CMake\share\cmake-3.25\Modules\GoogleTest.cmake". Instead, it shoud navigate to the code of the test. This also happens when clicking the "Go-To-Test" button. This forces me to navigate to the test manually which could take time.

I think this is a duplicate of https://github.com/microsoft/vscode-cmake-tools/issues/3362

This is not so much a bug but simply a missing feature when using `gtest_discover_tests()` as there's no way the discovery reports source file and line back to cmake and therefore the vscode cmake plugin can't pick it up.

**Comment by gcampbell-msft** [comment]
@bertbuchholz @yoavmil This issue should be fixed in the most recent official release, could you confirm and if it still reproduces, create a new issue? Closing this issue since it should be fixed by #3631

**Comment by yoavmil** [comment]
Actaully now nothing doesn't work out of the box.
Does cmake-tools actually come with cmake? because it cant find it. And after I install CMake and set PATH to it, it cant find Ninja.
